src/components/PostsList.tsx

```diff
@@ -1,86 +1,66 @@
 import React from 'react';
+import { Post } from '../types/Post';
+import cn from 'classnames';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[];
+  currPost: Post | null;
+  setModalWindow: (value: boolean) => void;
+  setCurrPost: (postId: Post | null) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  currPost,
+  setModalWindow,
+  setCurrPost,
+}) => {
+  const onOpenPost = (post: Post) => {
+    const isSamePost = currPost && currPost?.id === post.id;
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+    if (isSamePost) {
+      setCurrPost(null);
+    } else {
+      setCurrPost(post);
+      setModalWindow(false);
+    }
+  };
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
Consider removing the eslint-disable comment if it's not necessary. If the button lacks an associated label, ensure accessibility compliance by providing an appropriate label.